src/app/dataset-view/additional-components/metadata-component/components/form-components/schema-field/schema-field.component.ts

```diff
@@ -15,11 +38,92 @@ export interface SchemaType {
     styleUrls: ["./schema-field.component.sass"],
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class SchemaFieldComponent extends BaseField {
-    public displayedColumns: string[] = ["name", "type", "actions"];
-    public availableTypes = ["STRING", "BIGINT", "TIMESTAMP"];
-    private defaultType = "STRING";
+export class SchemaFieldComponent extends BaseField implements AfterViewInit {
+    private defaultType = "";
     @ViewChild(MatTable) table: MatTable<unknown>;
+    @ViewChildren("ngbTypeahed") ngbType: QueryList<NgbTypeahead>;
+    public focus$ = new Subject<SchemaType>();
+    public click$ = new Subject<SchemaType>();
+    private focusObservableList = Array<Subject<SchemaType>>();
+    private clickObservableList = Array<Subject<SchemaType>>();
+    private ngbTypeHeader: NgbTypeahead;
+
+    public displayedColumns: string[] = ["name", "type", "actions"];
```

Comment:
Once again, all constants should be UPPER_CASE and `readonly`

Reply:
Fixed